common/w_wad.cpp

```diff
@@ -170,6 +172,51 @@ std::string W_MD5(std::string filename)
 	return hash.str();
 }
 
+/*
+ * @brief Creates a fingerprint for a map via FarmHash.
+ *
+ * However, it encodes the fingerprint into a 16-byte array to be read later.
+ *
+ * @param lumpdata byte array pointer to the lump that needs to be fingerprinted.
+ * @param size of the byte array pointer.
+ */
+byte* W_FARMHASH(const byte* lumpdata, int length)
```

Comment:
This design is very error-prone, since you are allocating a byte buffer in the function contents (with `malloc` no less) but the name of the function does not indicate any allocation is going on, nor does the docstring.

What I would do instead is return the data in a struct containing one array member by value.  That allows you to pass back arrays without them decaying to pointers.  One day I'll make a OArray, but a hand-rolled struct will do for now.

Also, I prefer `W_FarmHash` since that's how the official hash name is capitalized - you might also want to add the size of the hash at the end in bits.